src/core/Rule.ml

```diff
@@ -906,15 +909,15 @@ let () = Printexc.register_printer opt_string_of_exn
 let visit_new_formula f formula =
   let bref = ref false in
   let rec visit_new_formula f formula =
-    match formula with
+    match formula.f with
     | P p -> f p ~inside:!bref
     | Anywhere (_, formula)
     | Inside (_, formula) ->
         Common.save_excursion bref true (fun () -> visit_new_formula f formula)
     | Not (_, x) -> visit_new_formula f x
     | Or (_, xs)
-    | And (_, { conjuncts = xs; _ }) ->
-        xs |> List.iter (visit_new_formula f)
+    | And (_, xs) ->
+        xs |> List.iter (fun formula -> visit_new_formula f formula)
```

Comment:
Nit picking, but in this particular case I think `(visit_new_formula f)` could be more readable.